### DISCUSSION TITLE
[11ty Navigation] Render menu items manually not recognizing "active" class

### DISCUSSION THREAD

**Discussion by sbootes** [11ty/eleventy#3086] [Uncategorized]
This is my first post on this discussion board. I hope I've followed the etiquette properly.  :-) 

I have a project that uses the navigation plugin to build a menu based on the markdown files in a folder. I'm using the render menu items manually code from the plugin documentation page with an added "else" condition in the logic. The menu tree builds properly but the flow control to mark a menu item as active seems to be ignored. If anyone has any ideas why the active class isn't being inserted, I'd really appreciate the help!

The markdown front matter looks like this:

```
---
eleventyNavigation:
  title: Overview
  url: overview.html
  order: 1
---
```

The template page is HTML based with liquid syntax:
```
<div class="collapse navbar-collapse" id="navbarText">
    {% assign navPages = collections.all | eleventyNavigation %}
    <ul class="navbar-nav ms-auto nav">
    {%- for entry in navPages %}
        <li {% if entry.url == page.url %} class="nav-item active" aria-current="page" {% else %} class="nav-item" {% endif %} >
            <a class="nav-link" href="{{ entry.url }}">{{ entry.title }}</a>
        </li>
    {%- endfor %}
    </ul>
 </div>
```

**Comment by Snapstromegon** [comment]
Without seeing the whole project it's a little hard for me to give you a direct answer. Have you tried not overwriting the url field and have you tried outputting the value of `entry.url` and `page.url` to check if the values are what you expect them to be?

**Reply by sbootes** [reply]
Thank you for your reply. I am building a site with 89 files over 14 folders and I need a common set of files in each folder. (The folder hierarchy was left out for clarity - but even at the root level, the active page is not picked up.) So, if I don't overwrite the URL field, I get one sub-folder for every page, which is entirely unworkable. The values for entry.url and page.url do appear correct.

**Reply by Snapstromegon** [reply]
Would it be possible to create maybe a reduced example as a git repo (if you can't share the original), so I can reproduce this and take a closer look? Everything I could do right now would be like throwing darts into the void.
